utils.py

```diff
@@ -70,26 +71,39 @@ class DataReader(object):
     def __init__(self):
         self.vocab = Vocab()
         self.data = None
+        self.cache_file = '/tmp/nlp-project-sentences-train.pickle'
         return
     
     def construct(self, path, vocab_size, sent_size):
         '''Load vocabulary, mapping words to tokens, then load corpus as tokens.
         '''
-        # intial pass over path to construct vocabulary 
-        self.vocab.construct(path, vocab_size)
+        if os.path.isfile(self.cache_file):
+            print('loading cached data from: %s' % self.cache_file)
+            with open(self.cache_file, 'rb') as f:
+                self.data = pickle.load(f)
+        else:
+            print('constructing data set')
+
+            # intial pass over path to construct vocabulary 
+            self.vocab.construct(path, vocab_size)
```

Comment:
The model only needs the tokens, but its nice to keep the words around for interpretation. We might need it for the sentence generation